reconcile/external_resources/factories.py

```diff
@@ -48,6 +48,8 @@ def register_factory(self, id: str, t: T) -> None:
         self._factories[id] = t
 
     def get_factory(self, id: str) -> T:
+        if id not in self._factories and "default" in self._factories:
+            return self._factories["default"]
```

Comment:
Should we make `default` explicit instead of a hard coded string? For example, `ObjectFactory` register all factories in `__init__`

```python
class ObjectFactory(Generic[T]):
    def __init__(self, factories, default_factory = None) -> None:
        self._factories = factories
        self._default_factory = default_factory

    def get_factory(self, id: str) -> T:
        if id not in self._factories and self._default_factory:
            return self._default_factory
        return self._factories[id]
```

init once and immutable

Reply:
I like it. Added.